### PR TITLE
Python 2.7 compatibility

### PR DESCRIPTION
**Pull request by psycofdj**
- few changes in language syntax
 - using unittest2 instead of legacy unittest
 - using future for metaclasses
 - testing the whole thing with tox
 - adds unittest for xtd.core.mixin
 - adds a few pytlint disable

### CODE REVIEW

**Comment by psycofdj** [comment]
@Anvil, @max-k  I'd love to have your feedback on this PR, especially the debian part. 
I'm not sure how to produce correctly both python-xtd and python3-xtd packages.

**Comment by max-k** [comment]
Looks good to me ;-)

**Comment by max-k** [comment]
You can probably remove anything after override_dh_auto_test: in debian/rules.

File: xtd/core/application.py [hunk 1]
```diff
@@ -214,10 +218,12 @@ def execute(self, p_argv=None):
     try:
       self.initialize()
     except ConfigError as l_error:
+      #pylint: disable=superfluous-parens
```
Comment (psycofdj) [inline xtd/core/application.py]: I'll give it a try, thx

**Pull request merged**